build.gradle

```diff
@@ -986,7 +986,14 @@ if (jdkVersion == '8') {
 }
 
 project(':iceberg-spark3') {
+  apply plugin: 'scala'
+
   sourceSets {
+    main {
+      // Use Scala Compiler for everything so we can do cross compilation within the module
+      scala.srcDirs = ['src/main/scala', 'src/main/java']
+      java.srcDirs = []
+    }
     // Compile test source against Spark 3.1 and main classes compiled against Spark 3.0
```

Comment:
Do we need to do anything with the spark31 sourceSet?

Reply:
Nope, we still use the "main" source set